luda-editor/luda-editor-client/src/app/editor/clip_editor/camera_clip_editor/image_browser/mod.rs

```diff
@@ -168,91 +188,25 @@ impl ImageBrowser {
         thumbnail_rect: XywhRect<f32>,
         image_filename_objects: &Vec<ImageFilenameObject>,
     ) -> Vec<BrowserItemProps> {
-        let mut iter = self.directory_key.split("-").filter(|s| !s.is_empty());
-        let character = iter.next();
-        let pose = iter.next();
-
-        if character.is_none() {
-            let mut characters = BTreeSet::new();
-            for filename_object in image_filename_objects {
-                characters.insert(&filename_object.character);
-            }
-            return characters
-                .into_iter()
-                .map(|character| {
-                    let filename_object = image_filename_objects
-                        .iter()
-                        .find(|filename_object| filename_object.character == *character)
-                        .unwrap();
-                    let key = format!("{}", character);
-                    BrowserItemProps {
-                        name: character.to_string(),
-                        thumbnail_url: filename_object.url.clone(),
-                        key: key.clone(),
-                        is_selected: self.selected_key == Some(key.clone()),
-                        item_size,
-                        thumbnail_rect,
-                    }
-                })
-                .collect();
-        };
-        let character = character.unwrap();
-
-        if pose.is_none() {
-            let mut poses = BTreeSet::new();
-            for filename_object in image_filename_objects
-                .iter()
-                .filter(|filename_object| filename_object.character == character)
-            {
-                poses.insert(&filename_object.pose);
-            }
-            return poses
-                .into_iter()
-                .map(|pose| {
-                    let filename_object = image_filename_objects
-                        .iter()
-                        .find(|filename_object| {
-                            filename_object.character == character && filename_object.pose == *pose
-                        })
-                        .unwrap();
-                    let key = format!("{}-{}", character, pose);
-                    BrowserItemProps {
-                        name: pose.to_string(),
-                        thumbnail_url: filename_object.url.clone(),
-                        key: key.clone(),
-                        is_selected: self.selected_key == Some(key),
-                        item_size,
-                        thumbnail_rect,
-                    }
-                })
-                .collect();
-        };
-        let pose = pose.unwrap();
+        let under_directory_items = image_filename_objects
```

Comment:
Previous code is very big and long, duplicated. I make it simpler using `impl` with good-named functions.